mindsdb/integrations/kafka/kafkadb.py

```diff
@@ -0,0 +1,172 @@
+import os
+import json
+import kafka
+
+from threading import Thread
+from mindsdb.utilities.config import STOP_THREADS_EVENT
+from mindsdb.utilities.log import log
+from mindsdb.integrations.base import Integration
+from mindsdb.streams.kafka.kafka_stream import KafkaStream
+from mindsdb.interfaces.storage.db import session, Stream, Configuration
+
+class KafkaConnectionChecker:
+    def __init__(self, **kwargs):
+        self.host = kwargs.get('host')
+        self.port = kwargs.get('port', 9092)
+
+    def _get_connection(self):
+        return kafka.KafkaAdminClient(bootstrap_servers=f"{self.host}:{self.port}")
+    def check_connection(self):
+        try:
+            client = self._get_connection()
+            client.close()
+            return True
+        except Exception:
+            return False
+
+
+class Kafka(Integration, KafkaConnectionChecker):
+    def __init__(self, config, name):
+        Integration.__init__(self, config, name)
+        intergration_info = self.config['integrations'][self.name]
+        self.host = intergration_info.get('host')
+        self.port = intergration_info.get('port', 9092)
+        self.control_topic_name = intergration_info.get('topic')
+        self.client = self._get_connection()
+        self.company_id = os.environ.get('MINDSDB_COMPANY_ID', None)
+        self.streams = {}
+        self.stop_event = STOP_THREADS_EVENT
+
+    def should_i_exist(self):
+        config_record = session.query(Configuration).filter_by(company_id=self.company_id).first()
+        if config_record is None:
+            return False
+        integrations = json.loads(config_record.data)["integrations"]
+        if self.name not in integrations:
+            return False
```

Comment:
This check is being done the wrong way, please check the configuration object directly since that already does the merging of the file config with the db config.

If we can't access the config here then we are doing something wrong and the control flow should be changed, but I think we can.

Reply:
Thanks. Changed on `Config()` usage